meorg_client/tests/test_client.py

```diff
@@ -62,38 +105,74 @@ def client():
 def test_login(client, mocker):
     """Test login."""
 
+    request = REQUEST_EXAMPLES['/login']
+    response = RESPONSE_EXAMPLES['/login']
+
     with mocker as rm:
-        client.login(**LOGIN_REQUEST)
+        client.login(**request)
 
         # Ensure the headers indicating successful login are set.
-        assert client.headers['X-User-Id'] == LOGIN_REPONSE['userId']
-        assert client.headers['X-Auth-Token'] == LOGIN_REPONSE['authToken']
+        assert client.headers['X-User-Id'] == response['userId']
+        assert client.headers['X-Auth-Token'] == response['authToken']
 
 
 def test_logout(client, mocker):
     """Test logout."""
 
+    request = REQUEST_EXAMPLES['/login']
+    response = RESPONSE_EXAMPLES['/logout']
+
     with mocker as rm: 
-        client.login(**LOGIN_REQUEST)
+        client.login(**request)
```

Comment:
`login()` sets some authentication headers on the client that get sent through with each subsequent request. `logout()` will invalidate the session on the server, but also implies that the headers be cleared on the client, as they would be useless. This basically tests to make sure that the headers are cleared, as there isn't really a way to test that the server has invalidated the session. This effectively resets the client to a blank state as if it was first instantiated.

I did try having a fixture for an authenticated client, but it needed to be authenticated in the mock context block, otherwise it attempts to actually authenticate with the actual base url.

Reply:
Does `client.headers` need sensible values for the tests to pass? If not we could set the headers attribute manually rather than calling `client.login()` when creating an authenticated client.

Reply:
Technically no, the headers just need to be set - it is the server itself that determines their validity.

Reply:
It is actually more code to set the headers outside of the client rather than recall the login routine... I might just leave it in.